tests/v2/validation/scaling/nodepools.go

```diff
@@ -0,0 +1,72 @@
+package scaling
+
+import (
+	"testing"
+
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
+	rke1 "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+	"github.com/rancher/rancher/tests/framework/extensions/scaling"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ProvisioningSteveResourceType = "provisioning.cattle.io.cluster"
+	defaultNamespace              = "fleet-default/"
+)
+
+// func TestScalingRKE2K3SNodePools(t *testing.T, client *rancher.Client, cluster *v1.SteveAPIObject, updatedCluster *apisV1.Cluster, machineConfig *v1.SteveAPIObject) {
+func TestScalingRKE2K3SWorkerNodePools(t *testing.T, client *rancher.Client, clusterID string, clustersConfig *provisioninginput.Config) {
+	cluster, err := client.Steve.SteveType(ProvisioningSteveResourceType).ByID(clusterID)
+	require.NoError(t, err)
+
+	updatedCluster := new(apisV1.Cluster)
+	err = v1.ConvertToK8sType(cluster, &updatedCluster)
+	require.NoError(t, err)
+
+	providerName := clustersConfig.Providers[0]
+	nodeProvider := scaling.GetClusterProvider(providerName)
+
+	machineConfig := updatedCluster.Spec.RKEConfig.MachinePools[0].NodeConfig.Name
+	machineConfigResp, err := client.Steve.SteveType(nodeProvider.MachineConfigPoolResourceSteveType).ByID(defaultNamespace + machineConfig)
+	require.NoError(t, err)
+
+	err, createdClusterResp, createdNodePoolCluster := machinepools.CreateNewWorkerMachinePool(client, cluster, updatedCluster, machineConfigResp, 3)
+	require.NoError(t, err)
+
+	err = pods.VerifyReadyDaemonsetPods(t, client, cluster)
+	require.NoError(t, err)
+
+	err, scaledClusterResp := machinepools.ScaleNewWorkerMachinePool(client, createdClusterResp, createdNodePoolCluster, 4)
```

Comment:
see my comment in machinepools on this function. seems more intuitive to send the number to scale by, not the total number of nodes in the pool.

Reply:
I think it's more intuitive to have the total number personally. However, if we all think the latter, I can look into that at a later point in time.

Reply:
Can you explain further Caleb? What do you mean?

Reply:
I don't think 'new' should be in the function name, and that it would make more sense to me to send the net change to the pool quantity instead of the new expected quantity to this function. To scale something, I shouldn't have to know the existing count, only how much to increase/decrease by IMO